IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.m

```diff
@@ -26,21 +26,327 @@
+{
+    if (!self.loading)
+    {
+        self.loading = YES;
+        if (_spinnerTimer)
+        {
+            [_spinnerTimer invalidate];
+        }
+        _spinnerTimer = [NSTimer scheduledTimerWithTimeInterval:2.0
+                                                         target:self
+                                                       selector:@selector(onStartLoadingIndicator:)
+                                                       userInfo:nil
+                                                        repeats:NO];
+        [_spinnerTimer setTolerance:0.3];
+    }
+}
+
+- (void)webView:(WKWebView *)webView didFinishNavigation:(null_unspecified WKNavigation *)navigation
+{
+    NSURL *url = webView.URL;
+    MSID_LOG_VERBOSE(self.context, @"-didFinishNavigation host: %@", [MSIDAuthority isKnownHost:url] ? url.host : @"unknown host");
+    MSID_LOG_VERBOSE_PII(self.context, @"-didFinishNavigation host: %@", url.host);
+    
+    [self stopSpinner];
+}
+
+- (void)webView:(WKWebView *)webView didFailNavigation:(null_unspecified WKNavigation *)navigation withError:(NSError *)error
+{
+    [self webAuthFailWithError:error];
+}
+
+- (void)webView:(WKWebView *)webView didReceiveAuthenticationChallenge:(NSURLAuthenticationChallenge *)challenge completionHandler:(ChallengeCompletionHandler)completionHandler
+{
+    NSString *authMethod = [challenge.protectionSpace.authenticationMethod lowercaseString];
+    
+    MSID_LOG_VERBOSE(self.context,
+                     @"%@ - %@. Previous challenge failure count: %ld",
+                     @"webView:didReceiveAuthenticationChallenge:completionHandler",
+                     authMethod, (long)challenge.previousFailureCount);
+    
+    [MSIDChallengeHandler handleChallenge:challenge
+                                  webview:webView
+                                  context:self.context
+                        completionHandler:completionHandler];
+}
+
+- (void)completeWebAuthWithURL:(NSURL *)endURL
+{
+    MSID_LOG_INFO(self.context, @"-completeWebAuthWithURL: %@", [MSIDAuthority isKnownHost:endURL] ? endURL.host : @"unknown host");
+    MSID_LOG_INFO_PII(self.context, @"-completeWebAuthWithURL: %@", endURL);
+    
+    [self endWebAuthWithURL:endURL error:nil];
+}
+
+// Authentication failed somewhere
+- (void)webAuthFailWithError:(NSError *)error
+{
+    if (self.complete)
+    {
+        return;
+    }
+    
+    if ([error.domain isEqualToString:NSURLErrorDomain] && NSURLErrorCancelled == error.code)
+    {
+        //This is a common error that webview generates and could be ignored.
+        //See this thread for details: https://discussions.apple.com/thread/1727260
+        return;
+    }
+    
+    // Ignore WebKitError 102 for OAuth 2.0 flow.
+    if ([error.domain isEqualToString:@"WebKitErrorDomain"] && error.code == 102)
+    {
+        return;
+    }
+    
+    [self stopSpinner];
+    
+    if([error.domain isEqual:@"WebKitErrorDomain"])
```

Comment:
confused about this, why just checked for WebKitErrorDomain on 102 specifically, which ended this method.
Why are we checking error domain again here?

Reply:
You are right, now I removed line 268-271 and added more comments here.